app/Policies/UserPolicy.php

```diff
@@ -114,7 +116,24 @@ public function viewAllApplications(User $user): bool
      */
     public function viewUnfinishedApplications(User $user): bool
     {
-        return $this->viewAllApplications($user);
+        return $user->hasRole([
+            Role::SECRETARY,
+            Role::DIRECTOR,
+            Role::STUDENT_COUNCIL => Role::STUDENT_COUNCIL_LEADERS,
+        ]);
+    }
+
+    /**
+     * Returns true if the user can finalize the application process.
+     * @param User $user
+     * @return bool
+     */
+    public function finalizeApplicationProcess(User $user): bool
+    {
+        return $user->hasRole([[
+            Role::SYS_ADMIN,
+            Role::STUDENT_COUNCIL => Role::PRESIDENT,
```

Comment:
I'd only say admins here... This cannot be undone

Reply:
Also maybe check here that the current date is at least the deadline date + 3 weeks.